### PR TITLE
Feature/first mutations and queries

### PR DESCRIPTION
**Pull request by ederivero**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Migrations have been added for schema changes.

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the new behavior?

- First queries and mutations for the current project with the new architecture

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

....

### CODE REVIEW

File: src/schema/mutations/index.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { GraphQLObjectType, GraphQLNonNull, GraphQLID } from "graphql"
+import { productMutations } from "./products.mutations"
+
+export default new GraphQLObjectType({
+  name: "Mutation",
```
Comment (fhereduardo90) [inline src/schema/mutations/index.ts]: use single quotes, also configure prettier and eslint to make sure that the whole project follows our best practices

File: src/schema/mutations/products.mutations.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { GraphQLFieldConfig, GraphQLNonNull } from "graphql"
+import { ProductService } from "../../services/products.service"
+import productInput from "../types/product.input"
+import { ProductType } from "../types/product.type"
+
+const createProduct: GraphQLFieldConfig<any, any> = {
```
Comment (fhereduardo90) [inline src/schema/mutations/products.mutations.ts]: in the future can we avoid using any?

File: src/schema/mutations/products.mutations.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  type: ProductType,
+  args: { input: { type: new GraphQLNonNull(productInput) } },
+  resolve: async function (_source, { input }) {
+    const product = await ProductService.createProduct(input)
+
+    return product
```
Comment (fhereduardo90) [inline src/schema/mutations/products.mutations.ts]: just return the promise, probably in the future you will need to use `class-transformer`

File: src/schema/queries/index.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { GraphQLObjectType } from "graphql"
```
Comment (fhereduardo90) [inline src/schema/queries/index.ts]: single quotes please

File: src/schema/queries/user.queries.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+import { UserDto } from "../../services/dtos/response/user.dto"
+import { UserService } from "../../services/users.service"
+import { UserType } from "../types/user.type"
+
+const currentUser: GraphQLFieldConfig<any, any> = {
+  type: UserType,
+  args: { userId: { type: new GraphQLNonNull(GraphQLString) } },
+  resolve: async function (_source, { userId }) {
+    const user = await UserService.findUser(userId)
+    return plainToClass(UserDto, user)
```
Comment (fhereduardo90) [inline src/schema/queries/user.queries.ts:12]: add a blank line

File: src/utils/index.ts [hunk 6]
```diff
@@ -0,0 +1,39 @@
+import { PaginationArgs } from "./dtos/request/pagination.args"
```
Comment (fhereduardo90) [inline src/utils/index.ts]: remember add test for each utilitiy

File: src/schema/mutations/index.ts (hunk 1 above)
Comment (ederivero) [reply src/schema/mutations/index.ts]: Done

File: src/schema/queries/index.ts (hunk 4 above)
Comment (ederivero) [reply src/schema/queries/index.ts]: Done

File: src/utils/index.ts (hunk 6 above)
Comment (ederivero) [reply src/utils/index.ts]: Added

File: src/schema/queries/user.queries.ts (hunk 5 above)
Comment (ederivero) [reply src/schema/queries/user.queries.ts:12]: Added

File: src/schema/mutations/products.mutations.ts (hunk 3 above)
Comment (ederivero) [reply src/schema/mutations/products.mutations.ts]: Done

File: src/schema/mutations/products.mutations.ts (hunk 2 above)
Comment (ederivero) [reply src/schema/mutations/products.mutations.ts]: Done

**Pull request merged**